test/TestUtilities/Test.Utility/Commands/ProjectTestHelpers.cs

```diff
@@ -212,37 +192,15 @@ private static PackageSpec WithProjectJsonTestRestoreMetadata(this PackageSpec s
         }
 
         /// <summary>
-        /// Creates a restore request, with the only source being the source from the <paramref name="pathContext"/>.
+        /// Creates a restore request for the first project in the <paramref name="projects"/> list. If <see cref="ProjectRestoreMetadata.Sources"/> has any values, it is used for creating the providers, otherwise <see cref="SimpleTestPathContext.PackageSource"/> from <paramref name="pathContext"/> will be used.
         /// </summary>
-        /// <param name="spec"></param>
-        /// <param name="pathContext"></param>
-        /// <param name="logger"></param>
-        /// <returns></returns>
-        public static TestRestoreRequest CreateRestoreRequest(PackageSpec spec, SimpleTestPathContext pathContext, ILogger logger)
+        public static TestRestoreRequest CreateRestoreRequest(SimpleTestPathContext pathContext, ILogger logger, params PackageSpec[] projects)
         {
-            var sources = new List<PackageSource> { new PackageSource(pathContext.PackageSource) };
-            var dgSpec = new DependencyGraphSpec();
-            dgSpec.AddProject(spec);
-            dgSpec.AddRestore(spec.RestoreMetadata.ProjectUniqueName);
-
-            return new TestRestoreRequest(spec, sources, pathContext.UserPackagesFolder, logger)
-            {
-                LockFilePath = Path.Combine(spec.FilePath, LockFileFormat.AssetsFileName),
-                DependencyGraphSpec = dgSpec,
-            };
-        }
-
-        public static TestRestoreRequest CreateRestoreRequest(PackageSpec projectToRestore, IEnumerable<PackageSpec> packageSpecsClosure, SimpleTestPathContext pathContext, ILogger logger)
-        {
-            var sources = new List<PackageSource> { new PackageSource(pathContext.PackageSource) };
-            var dgSpec = new DependencyGraphSpec();
-            dgSpec.AddProject(projectToRestore);
-            dgSpec.AddRestore(projectToRestore.RestoreMetadata.ProjectUniqueName);
-
-            foreach (var spec in packageSpecsClosure)
-            {
-                dgSpec.AddProject(spec);
-            }
+            DependencyGraphSpec dgSpec = GetDGSpecForFirstProject(projects);
+            var projectToRestore = projects[0];
+            var sources = projectToRestore.RestoreMetadata.Sources.Any() ?
+                       projectToRestore.RestoreMetadata.Sources.ToList() :
+                       new List<PackageSource> { new PackageSource(pathContext.PackageSource) };
```

Comment:
Thanks for fixing this!
So seems previously the sources is just a local path. I'm just curious why the previous test is able to show http warning.

Reply:
>  I'm just curious why the previous test is able to show http warning

Because it's using the package spec. 

Your implementation should use the RemoteDependencyProvider PackageSource